API/Data/Repository/UserRepository.cs

```diff
@@ -16,56 +18,75 @@ public class UserRepository : IUserRepository
     private readonly DataContext _context;
     private readonly IMapper _mapper;
     public UserRepository(
-      DataContext context,
-      IMapper mapper
-    )
-    {
+      	DataContext context,
+      	IMapper mapper
+    ) {
         _context = context;
         _mapper = mapper;
     }
 
     public async Task<MemberDTO> GetMemberAsync(string username)
     {
-      return await _context.Users
-        .Where(x => x.UserName == username)
-        .ProjectTo<MemberDTO>(_mapper.ConfigurationProvider)
-        .SingleOrDefaultAsync();
+		return await _context.Users
+			.Where(x => x.UserName == username)
+			.ProjectTo<MemberDTO>(_mapper.ConfigurationProvider)
+			.SingleOrDefaultAsync();
     }
 
-    public async Task<IEnumerable<MemberDTO>> GetMembersAsync()
+    public async Task<PagedList<MemberDTO>> GetMembersAsync(UserParams userParams)
     {
-      return await _context.Users
-        .ProjectTo<MemberDTO>(_mapper.ConfigurationProvider)
-        .ToListAsync();
+    	var query = _context.Users.AsQueryable();
+
+		query = query.Where(u => u.UserName != userParams.CurrentUsername);
+		query = query.Where(u => u.Gender == userParams.Gender);
+
+		var minDob = DateTime.Today.AddYears(- userParams.MaxAge - 1);
```

Comment:
Not an issue, just a hint: usually when use SQL Server as the Database, try to use DateTimeOffset type rather than the DateTime.